### PR TITLE
fix: Use Olm CR sample from CSV

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Use Olm CR sample from CSV

### What issues does this PR fix or reference?
None. Code enhancement.

### How to test this PR?
Install Che on the Openshift platform:

```bash
$ chectl server:start -n che -p openshift -m
```
Take a look on the CR:

```bash
$ oc get checluster eclipse-che  -n che -o  yaml
```

Check Che using url. 

And also make sure that operator installer still works too:

```bash
$ chectl server:start -n che2 -p openshift -m --installer operator
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: src/api/kube.ts [hunk 1]
```diff
@@ -1733,6 +1731,12 @@ export class KubeHelper {
     })
   }
 
+  async getCSV(csvName: string, namespace: string): Promise<ClusterServiceVersion | undefined> {
+    const csvs = await this.getClusterServiceVersions(namespace)
+    const csv = csvs.items.find(item => item.metadata.name === csvName)
+    return csv
```
Comment (tolusha) [inline src/api/kube.ts:1737]: return csvs.items.find(..)

File: src/tasks/installers/operator.ts [hunk 2]
```diff
@@ -149,6 +149,21 @@ export class OperatorTasks {
...
+      {
+        title: 'Prepare Eclipse Che cluster CR',
+        task: async (ctx: any, task: any) => {
+          const cheCluster = await kube.getCheCluster(flags.chenamespace)
+          if (cheCluster) {
+            ctx.cheClusterCreated = true
+            task.title = `${task.title}...It already exists..`
+          } else {
+            const yamlFilePath = flags['che-operator-cr-yaml'] === '' ? ctx.resourcesPath + 'crds/org_v1_che_cr.yaml' : flags['che-operator-cr-yaml']
+            ctx.CR = yaml.safeLoad(fs.readFileSync(yamlFilePath).toString())
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:161]: basically ctx.CR and ctx.cheClusterCreated are the same, right?

File: src/api/kube.ts [hunk 3]
```diff
@@ -1265,73 +1265,71 @@ export class KubeHelper {
     }
   }
 
-  async createCheClusterFromFile(filePath: string, flags: any, ctx: any, useDefaultCR: boolean): Promise<any> {
-    let yamlCr = this.safeLoadFromYamlFile(filePath)
-
+  async createCheCluster(CR: any, flags: any, ctx: any, useDefaultCR: boolean): Promise<any> {
```
Comment (mmorhun) [inline src/api/kube.ts:1268]: I am not sure that CR is a good name here. I think we should mention `che` or better `checluster`. What do you think about `CheClusterCR`?

**Review by mmorhun — APPROVED** [review]

File: src/tasks/installers/operator.ts (hunk 2 above)
Comment (AndrienkoAleksandr) [reply src/tasks/installers/operator.ts]: Yes, I removed unused `ctx.cheClusterCreated `

File: src/api/kube.ts (hunk 3 above)
Comment (AndrienkoAleksandr) [reply src/api/kube.ts]: Done

**Comment by AndrienkoAleksandr** [comment]
rebased

File: src/tasks/installers/olm.ts [hunk 4]
```diff
@@ -27,7 +28,6 @@ export class OLMTasks {
     const kube = new KubeHelper(flags)
     return new Listr([
       this.isOlmPreInstalledTask(command, kube),
-      copyOperatorResources(flags, command.config.cacheDir),
```
Comment (tolusha) [inline src/tasks/installers/olm.ts:30]: Why we don't need this anymore?

Comment (AndrienkoAleksandr) [reply src/tasks/installers/olm.ts:30]: Yes, correct.

File: src/commands/server/deploy.ts [hunk 5]
```diff
@@ -243,27 +239,22 @@ export default class Deploy extends Command {
...
 
-    if (flags['che-operator-cr-yaml']) {
-      const cheOperatorCrYamlPath = flags['che-operator-cr-yaml']
-      if (fs.existsSync(cheOperatorCrYamlPath)) {
-        const cr: any = yaml.safeLoad(fs.readFileSync(cheOperatorCrYamlPath).toString())
-        if (cr && cr.spec && cr.spec.server && cr.spec.server.tlsSupport === false) {
-          return false
-        }
-      }
+    const customCR = ctx.CustomCR
```
Comment (tolusha) [inline src/commands/server/deploy.ts:248]: pls. use `ctx.customCR` instead of `ctx.CustomCR`

**Review by tolusha — APPROVED** [review]

**Comment by AndrienkoAleksandr** [comment]
CI is finally green...

**Pull request merged**